scripts/ci/check_compliance.py

```diff
@@ -730,6 +736,45 @@ def check_disallowed_defconfigs(self, kconf):
                                "--perl-regexp", regex_socs, "--", ":soc",
                                cwd=ZEPHYR_BASE)
 
+        manifest = Manifest.from_file()
+        for project in manifest.get_projects([]):
+            if not manifest.is_active(project):
+                continue
+
+            if not project.is_cloned():
+                continue
+
+            module_path = PurePath(project.abspath)
+            module_yml = module_path.joinpath('zephyr/module.yml')
+
+            if not Path(module_yml).is_file():
+                module_yml = module_path.joinpath('zephyr/module.yaml')
+
+            if Path(module_yml).is_file():
+                with Path(module_yml).open('r', encoding='utf-8') as f:
+                    meta = yaml.load(f.read(), Loader=SafeLoader)
+
+                    if 'build' in meta and 'settings' in meta['build']:
+                        if 'board_root' in meta['build']['settings']:
+                            tmp_path = module_path.joinpath(meta['build']['settings']['board_root'])
```

Comment:
`board_path`? also tempted to suggest that this block (759:766) should go in its own function, the indentation here is brutal

Reply:
Have reworked into a function